src/Bicep.Local.Deploy/Extensibility/AzExtensibilityExtension.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Deployments.Core.Definitions;
+using Azure.Deployments.Engine.Host.Azure.ExtensibilityV2.Contract.Models;
+using Azure.Deployments.Extensibility.Contract;
+using Azure.Deployments.Extensibility.Data;
+using Azure.Deployments.Extensibility.Messages;
+using Microsoft.WindowsAzure.ResourceStack.Common.Extensions;
+using Microsoft.WindowsAzure.ResourceStack.Common.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Bicep.Local.Deploy.Extensibility;
+
+public class AzExtensibilityExtension : LocalExtensibilityExtension
+{
+    private readonly LocalExtensibilityHandler extensibilityHandler;
+
+    public AzExtensibilityExtension(LocalExtensibilityHandler extensibilityHandler)
+    {
+        this.extensibilityHandler = extensibilityHandler;
+    }
+
+    public override async Task<ResourceResponseBody> CreateOrUpdate(ResourceRequestBody request, CancellationToken cancellationToken)
+    {
+        switch (request.Type)
+        {
+            case "Microsoft.Resources/deployments":
+                {
+                    var template = request.Properties["template"]!.ToString();
+                    var parameters = new JObject
+                    {
+                        ["$schema"] = "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+                        ["contentVersion"] = "1.0.0.0",
+                        ["parameters"] = request.Properties["parameters"],
+                    };
+
+                    var result = await LocalDeployment.Deploy(extensibilityHandler, template, parameters.ToJson(), cancellationToken);
```

Comment:
Is `extensibilityHandler` used by `LocalDeployment`?

Reply:
Is used to register the service -> `service.AddSingleton()`

Reply:
Got it, it seems it's used by `LocalDeploymentEngineHost`.